Bottom: 2a23bb504e7919134c657861ae402ed215f628c2
Top:    64bb76c87124025e2a1a803435e23f18f0cc84d5
Author: Ian Kent <ikent@redhat.com>
Date:   2013-09-23 14:21:17 +0800

Refresh of debug.patch

---

diff --git a/lib/rpc_subs.c b/lib/rpc_subs.c
index 6ac8d32..1f35b01 100644
--- a/lib/rpc_subs.c
+++ b/lib/rpc_subs.c
@@ -345,17 +345,15 @@ static rpcprog_t rpc_getrpcbyname(const rpcprog_t program)
 	struct rpcent *entry;
 	unsigned int i;
 
-	if (table != NULL) {
-		pthread_mutex_lock(&rpcb_mutex);
-		for (i = 0; rpcb_pgmtbl[i] != NULL; i++) {
-			entry = getrpcbyname(rpcb_pgmtbl[i]);
-			if (entry) {
-				pthread_mutex_unlock(&rpcb_mutex);
-				return (rpcprog_t)entry->r_number;
-			}
+	pthread_mutex_lock(&rpcb_mutex);
+	for (i = 0; rpcb_pgmtbl[i] != NULL; i++) {
+		entry = getrpcbyname(rpcb_pgmtbl[i]);
+		if (entry) {
+			pthread_mutex_unlock(&rpcb_mutex);
+			return (rpcprog_t)entry->r_number;
 		}
-		pthread_mutex_unlock(&rpcb_mutex);
 	}
+	pthread_mutex_unlock(&rpcb_mutex);
 #endif
 	return program;
 }
